exporter/sumologicexporter/config.go

```diff
@@ -0,0 +1,130 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicexporter
+
+import (
+	"time"
+
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+// Config defines configuration for Sumo Logic exporter.
+type Config struct {
+	configmodels.ExporterSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
+
+	// Global options
+	// Unique URL generated for your HTTP Source.
+	// This is the address to send data to.
+	// url: "https://collectors.sumologic.com/receiver/v1/http/<UniqueHTTPCollectorCode>"
+	URL string `mapstructure:"url"`
+	// Option to enable compression (default true)
+	Compress bool `mapstructure:"compress"`
+	// Compression encoding format, either gzip or deflate (default gzip)
+	CompressEncoding CompressEncodingType `mapstructure:"compress_encoding"`
+	// Max HTTP request body size in bytes before compression (if applied).
+	// By default 1MB is recommended.
+	MaxRequestBodySize int `mapstructure:"max_request_body_size"`
+
+	// Logs related configuration
+	// Format to post logs into Sumo. (default json)
+	//   * text - Logs will appear in Sumo Logic in text format.
+	//   * json - Logs will appear in Sumo Logic in json format.
+	LogFormat LogFormatType `mapstructure:"log_format"`
+
+	// Metrics related configuration
+	// The format of metrics you will be sending, either graphite or carbon2 or prometheus (Default is carbon2)
+	MetricFormat MetricFormatType `mapstructure:"metric_format"`
+
+	// List of regexes for attributes which should be send as metadata
+	MetadataAttributes []string `mapstructure:"metadata_attributes"`
+
+	// Sumo specific options
+	// Desired source category.
+	// Useful if you want to override the source category configured for the source.
+	SourceCategory string `mapstructure:"source_category"`
+	// Desired source name.
+	// Useful if you want to override the source name configured for the source.
+	SourceName string `mapstructure:"source_name"`
+	// Desired host name.
+	// Useful if you want to override the source host configured for the source.
+	SourceHost string `mapstructure:"source_host"`
+	// Name of the client
+	Client string `mapstructure:"client"`
+
+	exporterhelper.TimeoutSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
+	exporterhelper.QueueSettings   `mapstructure:"sending_queue"`
+	exporterhelper.RetrySettings   `mapstructure:"retry_on_failure"`
```

Comment:
Consider to move these to the top close to the `ExporterSettings`. Keep kind of an order "standard embedded objects, custom objects"

Reply:
Done